sunpy/physics/solar_rotation.py

```diff
@@ -64,6 +64,10 @@ def calculate_solar_rotate_shift(mc, layer_index=0, **kwargs):
 
     # Calculate the rotations and the shifts
     for i, m in enumerate(mc):
+        # Skip the reference layer
```

Comment:
how did this end up here :grinning:

Reply:
I rebased this PR to include #3255 because the tweaking of transformations in this PR caused even more not-exactly-zero numerical-precision bugs that resulted in test failures.  At the time, all tests were green.  (At the time of this writing, tests are not green for an unrelated reason.)

Reply:
As an aside, I intended to leave #3255 for someone else to do, but I was "forced" to do it myself so that the tests would pass here.  Apparently if you want to squash some pesky bug in SunPy, just find some way for it to impact coordinates...